The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12216381176/job/34079421995) [(plain)](https://github.com/rust-lang/rust/commit/21b7d68934f61d64807eafdc909bacee1b4ed8a1/checks/34079421995/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 3.020 Building wheels for collected packages: reuse
#13 3.021   Building wheel for reuse (pyproject.toml): started
#13 3.271   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.272   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.272   Stored in directory: /tmp/pip-ephem-wheel-cache-sllpv1x6/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.275 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.680 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.680 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.219 Collecting virtualenv
#13 4.219 Collecting virtualenv
#13 4.285   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.608      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 13.3 MB/s eta 0:00:00
#13 4.653 Collecting distlib<1,>=0.3.7
#13 4.666   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.687      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 25.4 MB/s eta 0:00:00
#13 4.726 Collecting platformdirs<5,>=3.9.1
#13 4.739   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.781 Collecting filelock<4,>=3.12.2
#13 4.795   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.876 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.059 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 5.1s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      300992 kB
DirectMap2M:     7038976 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
Diff in /checkout/library/core/src/intrinsics/mod.rs:3577:
     unimplemented!()
 }
 
-    /// Rust's "try catch" construct for unwinding. Invokes the function pointer `try_fn` with the
-    /// data pointer `data`, and calls `catch_fn` if unwinding occurs while `try_fn` runs.
-    /// `catch_fn` must not unwind.
-    ///
-    /// The third argument is a function called if an unwind occurs (both Rust `panic` and foreign
-    /// The third argument is a function called if an unwind occurs (both Rust `panic` and foreign
-    /// unwinds). This function takes the data pointer and a pointer to the target- and
-    /// runtime-specific exception object that was caught.
-    /// Note that in the case of a foreign unwinding operation, the exception object data may not be
-    /// safely usable from Rust, and should not be directly exposed via the standard library. To
-    /// prevent unsafe access, the library implementation may either abort the process or present an
-    /// opaque error type to the user.
-    /// opaque error type to the user.
-    ///
-    /// For more information, see the compiler's source, as well as the documentation for the stable
-    /// version of this intrinsic, `std::panic::catch_unwind`.
-    #[rustc_intrinsic]
-    #[rustc_intrinsic_must_be_overridden]
-    #[rustc_nounwind]
-    pub unsafe fn catch_unwind(_try_fn: fn(*mut u8), _data: *mut u8, _catch_fn: fn(*mut u8, *mut u8)) -> i32 {
-        unreachable!() 
-    }
+/// Rust's "try catch" construct for unwinding. Invokes the function pointer `try_fn` with the
+/// data pointer `data`, and calls `catch_fn` if unwinding occurs while `try_fn` runs.
+/// `catch_fn` must not unwind.
+///
+/// The third argument is a function called if an unwind occurs (both Rust `panic` and foreign
+/// The third argument is a function called if an unwind occurs (both Rust `panic` and foreign
+/// unwinds). This function takes the data pointer and a pointer to the target- and
+/// runtime-specific exception object that was caught.
+/// Note that in the case of a foreign unwinding operation, the exception object data may not be
+/// safely usable from Rust, and should not be directly exposed via the standard library. To
+/// safely usable from Rust, and should not be directly exposed via the standard library. To
+/// prevent unsafe access, the library implementation may either abort the process or present an
+/// opaque error type to the user.
+/// For more information, see the compiler's source, as well as the documentation for the stable
+/// For more information, see the compiler's source, as well as the documentation for the stable
+/// version of this intrinsic, `std::panic::catch_unwind`.
+#[rustc_intrinsic]
+#[rustc_intrinsic_must_be_overridden]
+#[rustc_nounwind]
+pub unsafe fn catch_unwind(
+    _try_fn: fn(*mut u8),
+    _data: *mut u8,
+    _catch_fn: fn(*mut u8, *mut u8),
+) -> i32 {
+}
 
 
-    /// Emits a `nontemporal` store, which gives a hint to the CPU that the data should not be held
-    /// in cache. Except for performance, this is fully equivalent to `ptr.write(val)`.
-    ///
-    /// Not all architectures provide such an operation. For instance, x86 does not: while `MOVNT`
-    /// exists, that operation is *not* equivalent to `ptr.write(val)` (`MOVNT` writes can be reordered
-    /// in ways that are not allowed for regular writes).
-    #[rustc_intrinsic]
-    #[rustc_intrinsic_must_be_overridden]
-    #[rustc_nounwind]
-    pub unsafe fn nontemporal_store<T>(_ptr: *mut T, _val: T) {
-        unreachable!() 
-
-
+/// Emits a `nontemporal` store, which gives a hint to the CPU that the data should not be held
+/// in cache. Except for performance, this is fully equivalent to `ptr.write(val)`.
+///
+/// Not all architectures provide such an operation. For instance, x86 does not: while `MOVNT`
+/// exists, that operation is *not* equivalent to `ptr.write(val)` (`MOVNT` writes can be reordered
+/// in ways that are not allowed for regular writes).
+#[rustc_intrinsic]
+#[rustc_intrinsic_must_be_overridden]
+#[rustc_nounwind]
+pub unsafe fn nontemporal_store<T>(_ptr: *mut T, _val: T) {
+}
 
 
 /// See documentation of `<*const T>::offset_from` for details.
 #[rustc_intrinsic_const_stable_indirect]
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/tests/run-make/lto-no-link-whole-rlib/main.rs" "/checkout/library/core/src/unit.rs" "/checkout/tests/run-make/lto-no-link-whole-rlib/lib2.rs" "/checkout/tests/run-make/lto-no-link-whole-rlib/lib1.rs" "/checkout/tests/run-make/lto-no-link-whole-rlib/rmake.rs" "/checkout/library/core/src/tuple.rs" "/checkout/library/core/src/intrinsics/simd.rs" "/checkout/library/core/src/intrinsics/mir.rs" "/checkout/library/core/src/intrinsics/mod.rs" "/checkout/library/core/src/clone/uninit.rs" "/checkout/library/core/src/panicking.rs" "/checkout/library/core/src/arch.rs" "/checkout/library/core/src/cell/lazy.rs" "/checkout/library/core/src/cell/once.rs" "/checkout/library/core/src/internal_macros.rs" "/checkout/library/core/src/primitive_docs.rs" "/checkout/library/core/src/hash/sip.rs" "/checkout/library/core/src/hash/mod.rs" "/checkout/library/core/src/io/borrowed_buf.rs" "/checkout/library/core/src/io/mod.rs" "/checkout/library/core/src/range/legacy.rs" "/checkout/library/core/src/range/iter.rs" "/checkout/library/core/src/macros/mod.rs" "/checkout/library/core/src/hint.rs" "/checkout/library/core/src/char/methods.rs" "/checkout/library/core/src/char/decode.rs" "/checkout/library/core/src/char/convert.rs" "/checkout/library/core/src/char/mod.rs" "/checkout/library/core/src/result.rs" "/checkout/library/core/src/str/lossy.rs" "/checkout/library/core/src/str/traits.rs" "/checkout/library/core/src/str/error.rs" "/checkout/library/core/src/str/iter.rs" "/checkout/library/core/src/str/pattern.rs" "/checkout/library/core/src/str/validations.rs" "/checkout/library/core/src/str/count.rs" "/checkout/library/core/src/str/converts.rs" "/checkout/library/core/src/str/mod.rs" "/checkout/library/core/src/random.rs" "/checkout/library/core/src/ub_checks.rs" "/checkout/library/core/src/iter/sources/from_coroutine.rs" "/checkout/library/core/src/iter/sources/empty.rs" "/checkout/library/core/src/iter/sources/from_fn.rs" "/checkout/library/core/src/iter/sources/once.rs" "/checkout/library/core/src/iter/sources/successors.rs" "/checkout/library/core/src/iter/sources/repeat_n.rs" "/checkout/library/core/src/iter/sources/repeat_with.rs" "/checkout/library/core/src/iter/sources/repeat.rs" "/checkout/library/core/src/iter/sources/once_with.rs" "/checkout/library/core/src/iter/traits/iterator.rs" "/checkout/library/core/src/iter/traits/collect.rs" "/checkout/library/core/src/iter/traits/unchecked_iterator.rs" "/checkout/library/core/src/iter/traits/exact_size.rs" "/checkout/library/core/src/iter/traits/double_ended.rs" "/checkout/library/core/src/iter/traits/accum.rs" "/checkout/library/core/src/iter/traits/mod.rs" "/checkout/library/core/src/iter/traits/marker.rs" "/checkout/library/core/src/iter/sources.rs" "/checkout/library/core/src/iter/range.rs" "/checkout/tests/run-make/compile-stdin/rmake.rs" "/checkout/tests/run-make/crate-loading-crate-depends-on-itself/foo-current.rs" "/checkout/tests/run-make/crate-loading-crate-depends-on-itself/foo-prev.rs" "/checkout/tests/run-make/crate-loading-crate-depends-on-itself/rmake.rs" "/checkout/library/core/src/task/mod.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Sat Dec  7 22:00:01 UTC 2024
  network time: Sat, 07 Dec 2024 22:00:01 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/intrinsics/mod.rs
+++ b/library/core/src/intrinsics/mod.rs
@@ -3577,34 +3577,44 @@ pub const fn discriminant_value<T>(_v: &T) -> <T as DiscriminantKind>::Discrimin
     unimplemented!()
 }
 
-extern "rust-intrinsic" {
-    /// Rust's "try catch" construct for unwinding. Invokes the function pointer `try_fn` with the
-    /// data pointer `data`, and calls `catch_fn` if unwinding occurs while `try_fn` runs.
-    ///
-    /// `catch_fn` must not unwind.
-    ///
-    /// The third argument is a function called if an unwind occurs (both Rust `panic` and foreign
-    /// unwinds). This function takes the data pointer and a pointer to the target- and
-    /// runtime-specific exception object that was caught.
-    ///
-    /// Note that in the case of a foreign unwinding operation, the exception object data may not be
-    /// safely usable from Rust, and should not be directly exposed via the standard library. To
-    /// prevent unsafe access, the library implementation may either abort the process or present an
-    /// opaque error type to the user.
-    ///
-    /// For more information, see the compiler's source, as well as the documentation for the stable
-    /// version of this intrinsic, `std::panic::catch_unwind`.
-    #[rustc_nounwind]
-    pub fn catch_unwind(try_fn: fn(*mut u8), data: *mut u8, catch_fn: fn(*mut u8, *mut u8)) -> i32;
+/// Rust's "try catch" construct for unwinding. Invokes the function pointer `try_fn` with the
+/// data pointer `data`, and calls `catch_fn` if unwinding occurs while `try_fn` runs.
+///
+/// `catch_fn` must not unwind.
+///
+/// The third argument is a function called if an unwind occurs (both Rust `panic` and foreign
+/// unwinds). This function takes the data pointer and a pointer to the target- and
+/// runtime-specific exception object that was caught.
+///
+/// Note that in the case of a foreign unwinding operation, the exception object data may not be
+/// safely usable from Rust, and should not be directly exposed via the standard library. To
+/// prevent unsafe access, the library implementation may either abort the process or present an
+/// opaque error type to the user.
+///
+/// For more information, see the compiler's source, as well as the documentation for the stable
+/// version of this intrinsic, `std::panic::catch_unwind`.
+#[rustc_intrinsic]
+#[rustc_intrinsic_must_be_overridden]
+#[rustc_nounwind]
+pub unsafe fn catch_unwind(
+    _try_fn: fn(*mut u8),
+    _data: *mut u8,
+    _catch_fn: fn(*mut u8, *mut u8),
+) -> i32 {
+    unreachable!()
+}
 
-    /// Emits a `nontemporal` store, which gives a hint to the CPU that the data should not be held
-    /// in cache. Except for performance, this is fully equivalent to `ptr.write(val)`.
-    ///
-    /// Not all architectures provide such an operation. For instance, x86 does not: while `MOVNT`
-    /// exists, that operation is *not* equivalent to `ptr.write(val)` (`MOVNT` writes can be reordered
-    /// in ways that are not allowed for regular writes).
-    #[rustc_nounwind]
-    pub fn nontemporal_store<T>(ptr: *mut T, val: T);
+/// Emits a `nontemporal` store, which gives a hint to the CPU that the data should not be held
+/// in cache. Except for performance, this is fully equivalent to `ptr.write(val)`.
+///
+/// Not all architectures provide such an operation. For instance, x86 does not: while `MOVNT`
+/// exists, that operation is *not* equivalent to `ptr.write(val)` (`MOVNT` writes can be reordered
+/// in ways that are not allowed for regular writes).
+#[rustc_intrinsic]
+#[rustc_intrinsic_must_be_overridden]
+#[rustc_nounwind]
+pub unsafe fn nontemporal_store<T>(_ptr: *mut T, _val: T) {
+    unreachable!()
 }
 
 /// See documentation of `<*const T>::offset_from` for details.

